File: tests/php/classes/class_inheritance.php
<?php
// vybe-test: php/classes/class_inheritance
// origin: languages/php/tests/php/test_classes.rs
// vybe-test-mode: compile

class Animal { public $name; public function __construct($n) { $this->name = $n; } } class Cat extends Animal { public function speak() { return $this->name . ' says Meow'; } } $c = new Cat('Whiskers'); echo $c->speak();

File: tests/dart/dart_apis/class_inheritance.dart
// vybe-test: dart/dart_apis/class_inheritance
// origin: languages/dart/tests/dart/test_dart_apis.rs

class Animal { String name; Animal(this.name); } class Dog extends Animal { Dog(String name) : super(name); String speak() => name + ' barks'; }

void main() {}
